### PR TITLE
Add donate modal

### PR DESCRIPTION
**Pull request by SwenVanZanten**
schermafbeelding 2018-07-17 om 07 49 20

### CODE REVIEW

File: src/components/modal/DonatePanel.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import * as React from 'react'
+import { Col, Row } from 'reactstrap'
+import Modal from '../Modal'
+import { inject, observer } from 'mobx-react'
+import Gift from 'react-material-icon-svg/dist/GiftIcon'
+import i18nReact from 'i18n-react'
```
Comment (marpme) [inline src/components/modal/DonatePanel.tsx]: Please use the new Tranlastion modules (`react-i18next`)

File: src/components/modal/DonatePanel.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    return 'NL'
+  }
+
+  getBalance() {
+    return 4434
+  }
+
+  getPrice() {
+    return 333
+  }
```
Comment (marpme) [inline src/components/modal/DonatePanel.tsx]: Do we really need the functions above?

**Review by marpme — CHANGES_REQUESTED** [review]

File: src/components/modal/DonatePanel.tsx (hunk 2 above)
Comment (SwenVanZanten) [reply src/components/modal/DonatePanel.tsx]: Dude it's a wippy 😛

File: src/components/modal/DonatePanel.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      >
+        <p className="donate-description donate-description-larger">
+          At Verge we are always striving to achieve new heights and deliver products
+          or software implementations that are requested by the community.
+        </p>
+        <p  className="donate-description">
+          Listening to our community members and taking in feedback is not only
+          one of the core founding principles of Verge,
+          but it is inherent in our open-source nature.
+        </p>
```
Comment (marpme) [inline src/components/modal/DonatePanel.tsx]: Those sentences should also be translated, let's wrap it for now and leave it english only. We will care about translations later on.

**Review by marpme — CHANGES_REQUESTED** [review]

**Comment by marpme** [comment]
MOAR REJECTION.

**Comment by SwenVanZanten** [comment]
schermafbeelding 2018-07-17 om 13 56 20
@marpme haha it's a fucking WIP 😆 
You're reviewing the shit out of a WIP.
Go do your exams 😝

**Comment by marpme** [comment]
I'm done, so fucking done. hahaha

**Comment by SwenVanZanten** [comment]
schermafbeelding 2018-07-17 om 23 29 24


I've added some components to reduce code duplication.
I've copied the `sendTransaction` method though.
In the future we should abstract that too.

**Comment by marpme** [comment]
I would love to try to abstract that send Transaction method, do you think that this is possible with given structure?

**Comment by SwenVanZanten** [comment]
Yes of course! 😃 I think you should create a class with you feed the amount and to address.
Then update the status with a listener and the ending status with a listener.
Or something like that.

**Comment by marpme** [comment]
Promise: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise

Well, we could use that, so that you can subscribe certain events.

**Comment by SwenVanZanten** [comment]
👍 cool.
But do you want to do that in this branch?
Or make new branch after this is merged

**Comment by marpme** [comment]
Merged gonna use a new branch and file a new issue for this

**Pull request merged**